src/components/select/bl-select.ts

```diff
@@ -0,0 +1,402 @@
+   */
+  @property({ type: String, attribute: 'invalid-text' })
+  customInvalidText?: string;
+
+  /* Declare internal reactive properties */
+  @state()
+  private _isMenuOpen = false;
+
+  @state()
+  private _selectedOptions: ISelectOption[] = [];
+
+  @state()
+  private _additionalSelectedOptionCount = 0;
+
+  @state()
+  private _isInvalid = false;
+
+  @query('.selected-options')
+  private _selectedOptionsContainer!: HTMLElement;
+
+  @queryAll('.selected-options li')
+  private _selectedOptionsItems!: Array<HTMLElement>;
+
+  @query('.select-menu')
+  private _selectMenu: HTMLElement;
+
+  @query('.select-input')
+  private _selectInput: HTMLElement;
+
+  @event('bl-select') private _onBlSelect: EventDispatcher<ISelectOption[]>;
+
+  private _connectedOptions: BlSelectOption[] = [];
+
+  get options() {
+    return this._connectedOptions;
+  }
+
+  get isMenuOpen() {
+    return this._isMenuOpen;
+  }
+
+  get selectedOptions() {
+    return this._selectedOptions;
+  }
+
+  get additionalSelectedOptionCount() {
+    return this._additionalSelectedOptionCount;
+  }
+
+  get isInvalid() {
+    return this._isInvalid;
+  }
+
+  private _clickOutsideHandler = (event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+
+    if (!this.contains(target) && this._isMenuOpen) {
+      this._isMenuOpen = false;
+      this._checkRequired();
+    }
+  };
+
+  private _showSelectMenu() {
+    const defaultMaxHeight = parseInt(
+      getComputedStyle(this._selectMenu)
+        .getPropertyValue('--bl-select-menu-height')
+        .replace('px', '')
+    );
+
+    computePosition(this._selectInput, this._selectMenu, {
+      placement: 'bottom',
+      strategy: 'absolute',
+      middleware: [
+        flip(),
+        offset(8),
+        size({
+          apply(args: MiddlewareArguments & { availableHeight: number }) {
+            Object.assign(args.elements.floating.style, {
+              maxHeight: `${
+                args.availableHeight > defaultMaxHeight ? defaultMaxHeight : args.availableHeight
+              }px`,
+            });
+          },
+        }),
+      ],
+    }).then(({ x, y }) => {
+      this._selectMenu.style.setProperty('--left', `${x}px`);
+      this._selectMenu.style.setProperty('--top', `${y}px`);
+    });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    document.addEventListener('click', this._clickOutsideHandler);
+  }
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    document.removeEventListener('click', this._clickOutsideHandler);
+  }
+
+  inputTemplate() {
+    const inputSelectedOptions = html`<ul class="selected-options">
+      ${this._selectedOptions.map(item => html`<li>${item.text}</li>`)}
+    </ul>`;
+    const _selectedItemCount = this._additionalSelectedOptionCount
+      ? html`<span>+${this._additionalSelectedOptionCount}</span>`
+      : null;
+    const removeIcon = this._isRemoveIconVisible
+      ? html`<bl-icon
```

Comment:
We should use `bl-button` for icon only buttons. secondary text button with an `icon` and `label` attribute.

Reply:
I tried with these attributes but I couldn't get the exact appearance 😿

Reply:
Let me also have a look for that.